src/resources/2023/BurningLeaves.ts

```diff
@@ -0,0 +1,91 @@
+import { Item, Monster, itemAmount, visitUrl } from "kolmafia";
+import { haveInCampground } from "../../lib";
+import { get } from "../../property";
+import { $item, $monster } from "../../template-string";
+
+const item = $item`A Guide to Burning Leaves`;
+
+export const burnItem: Map<Item, number> = new Map([
+  [$item`autumnic bomb`, 37],
+  [$item`impromptu torch`, 42],
+  [$item`flaming fig leaf`, 43],
+  [$item`smoldering drape`, 44],
+  [$item`distilled resin`, 50],
+  [$item`autumnal aegis`, 66],
+  [$item`lit leaf lasso`, 69],
+  [$item`forest canopy bed`, 74],
+  [$item`autumnic balm`, 99],
+  [$item`day shortener`, 222],
+  [$item`coping juice`, 1111],
+  [$item`smoldering leafcutter ant egg`, 6666],
+  [$item`super-heated leaf`, 11111],
+]);
+
+export const burnMonster: Map<Monster, number> = new Map([
+  [$monster`flaming leaflet`, 11],
+  [$monster`flaming monstera`, 111],
+  [$monster`leaviathan`, 666],
+]);
+
+/**
+ * @returns Whether or not we currently `have` the GuidetoBurningLeaves
+ */
+export function have(): boolean {
+  return haveInCampground(item);
+}
+
+/**
+ * @returns The number of leaves we have remaining
+ */
+export function numberOfLeaves(): number {
+  return itemAmount($item`inflammable leaf`);
+}
+
+function visitLeaves() {
+  visitUrl("campground.php?preaction=leaves");
+}
+
+/**
+ * @param leaves Burns the desired number of leaves
+ * @returns whether we could burn that number of leaves
+ */
+export function burnLeaves(leaves: number): boolean {
+  if (leaves > numberOfLeaves()) {
+    visitLeaves();
+    visitUrl(`choice.php?pwd&whichchoice=1510&leaves=${leaves}`);
+    return true;
+  }
+  return false;
+}
+
+/**
+ * @param th determines which thing (Monster or Item) we want to burn leaves for
+ * @returns Whether we can burn for that thing; or burns for that thing
+ */
+export function burnFor(th: Monster | Item): boolean {
+  if (th instanceof Item) {
```

Comment:
You don't need to check `instanceof `. What you get from burnItem will be `number | undefined` and then checking that it's not undefined will narrow it to `number`. So, both Items and Monsters behave the same way.

Reply:
There _was_ actually a typing issue here, but it had to do with having the `Map<Monster | Item, number>` as a type description for the variable instead of applying the type arg to the Map constructor, making typescript's difficulty with tuples come into play.

That being said, you're right that instanceof isnt' the best solution. But it was a weird one!